src/openforms/authentication/contrib/digid_eherkenning_oidc/views.py

```diff
@@ -43,6 +43,11 @@ def get(self, request):
         if not next_url:
             raise DisallowedRedirect
 
+        # This is needed for the next steps in order to redirect the user to the right place again.
+        # When a user is authenticated and cancels the login flow, the next url (oidc_login_next)
+        # has been removed from the session (the user is logged out-session flushed).
+        self.request.session["of_redirect_next"] = next_url
```

Comment:
Is it necessary to have this extra param in the session? The overwritten `get` of the `OIDCAuthenticationCallbackView` would still have `oidc_login_next` added by the `OIDCAuthenticationRequestView` (https://github.com/mozilla/mozilla-django-oidc/blob/1.2.4/mozilla_django_oidc/views.py#L179) when creating the `self._redirect_next`?

Reply:
yes, I specifically requested this :laughing: 

The reasoning is that *we* should not overload the `oidc_login_next` parameter meaning, it's actually wrong in the first place (IMO) that the failure URL makes use of the `self.success_url`. This is particularly dangerous if the upstream library decides to use this session key in other security sensitive areas (hypothetical example - checking if someone is considered "logged in" by checking the presence of this key).

Either way, this particular session key is private API of the mozilla-django-oidc library and we should *not* (ab)use that, the safe approach is adding our own semantics that are similar, but isolated.